src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/SemanticTokensOrSemanticTokensEdits.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Models
+{
+    internal struct SemanticTokensOrSemanticTokensEdits
```

Comment:
VS has a `SumType`, the converter that had to write to make that work involves Exceptions as control-flow, and I wasn't able to re-write it in such a way as to avoid that. Given that I'd prefer to use this method, which is a bit more of an upfront cost but saves us in that respect. It's also what O# did.